patches_for_WebRTC_org/m80/0007.4-Adding-video-profiles-capabilities-to-the-video-capt.patch

```diff
@@ -0,0 +1,1089 @@
++    _COM_Outptr_ ::ABI::Windows::Media::Capture::
++        IMediaCaptureInitializationSettings** pp_initialization_settings);
++
++HRESULT GetMediaCaptureWithInitSettings(
++    _In_ ::ABI::Windows::Media::Capture::IMediaCaptureInitializationSettings*
++        initialization_settings,
++    _COM_Outptr_ ::ABI::Windows::Media::Capture::IMediaCapture** media_capture);
+ 
+ HRESULT WaitForAsyncAction(
+-    ::Microsoft::WRL::ComPtr<ABI::Windows::Foundation::IAsyncAction>
+-        async_action);
++    _In_ ::ABI::Windows::Foundation::IAsyncAction* async_action);
+ 
+ template <typename T>
+ HRESULT WaitForAsyncOperation(
+-    ::Microsoft::WRL::ComPtr<ABI::Windows::Foundation::IAsyncOperation<T>>&
+-        async_op) {
++    _In_ ::ABI::Windows::Foundation::IAsyncOperation<T>* p_async_op) {
+   using ::ABI::Windows::Foundation::AsyncStatus;
+   using ::ABI::Windows::Foundation::IAsyncInfo;
+   using ::ABI::Windows::Foundation::IAsyncOperation;
+@@ -57,6 +64,7 @@ HRESULT WaitForAsyncOperation(
+   HRESULT hr = S_OK;
+   HRESULT hr_async_error = S_OK;
+   const DWORD timeout_ms = 2000;
++  ComPtr<IAsyncOperation<T>> async_op(p_async_op);
+   ComPtr<IAsyncInfo> async_info;
+   Event event_completed;
+ 
+@@ -73,9 +81,11 @@ HRESULT WaitForAsyncOperation(
+ 
+   // Creates the Event to be used to block and suspend until the async
+   // operation finishes.
+-  event_completed =
+-      Event(::CreateEventEx(nullptr, nullptr, CREATE_EVENT_MANUAL_RESET,
+-                            WRITE_OWNER | EVENT_ALL_ACCESS));
++  if (SUCCEEDED(hr)) {
++    event_completed =
++        Event(::CreateEventEx(nullptr, nullptr, CREATE_EVENT_MANUAL_RESET,
++                              WRITE_OWNER | EVENT_ALL_ACCESS));
++  }
+   if (SUCCEEDED(hr)) {
+     hr = event_completed.IsValid() ? S_OK : E_HANDLE;
+   }
+diff --git a/modules/video_capture/windows/video_capture_winrt.cc b/modules/video_capture/windows/video_capture_winrt.cc
+index a41fccfdb1..a0a323e64b 100644
+--- a/modules/video_capture/windows/video_capture_winrt.cc
++++ b/modules/video_capture/windows/video_capture_winrt.cc
+@@ -52,6 +52,8 @@ using ::ABI::Windows::Media::Capture::IMediaCapture;
+ using ::ABI::Windows::Media::Capture::IMediaCapture5;
+ using ::ABI::Windows::Media::Capture::IMediaCaptureInitializationSettings;
+ using ::ABI::Windows::Media::Capture::IMediaCaptureInitializationSettings5;
++using ::ABI::Windows::Media::Capture::IMediaCaptureVideoProfile;
++using ::ABI::Windows::Media::Capture::IMediaCaptureVideoProfileMediaDescription;
+ using ::ABI::Windows::Media::Capture::MediaCaptureMemoryPreference;
+ using ::ABI::Windows::Media::Capture::MediaStreamType;
+ using ::ABI::Windows::Media::Capture::StreamingCaptureMode;
+@@ -145,9 +147,9 @@ VideoCaptureWinRTInternal::~VideoCaptureWinRTInternal() {
+   }
+ }
+ 
+-HRESULT VideoCaptureWinRTInternal::InitCamera(const wchar_t* device_unique_id) {
++HRESULT VideoCaptureWinRTInternal::InitCamera(
++    const wchar_t* /*device_unique_id FIXME */) {
```

Comment:
This is a regression?

Reply:
Thanks for catching this one. It was a missing refactoring. ``VideoCaptureWinRTInternal::InitCamera`` serves no purpose anymore. It's initialization code has been moved to ``StartCapture`` and it's unnitialization code has been moved to ``VideoCaptureWinRTInternal::StopCamera``.